Mathlib/Algebra/Ring/Int/Defs.lean

```diff
@@ -0,0 +1,75 @@
+/-
+Copyright (c) 2016 Jeremy Avigad. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jeremy Avigad
+-/
+import Mathlib.Algebra.CharZero.Defs
+import Mathlib.Algebra.Group.Int
+import Mathlib.Algebra.Ring.Defs
+
+/-!
+# The integers are a ring
+
+This file contains the commutative ring instance on `ℤ`.
+
+See note [foundational algebra order theory].
+-/
+
+assert_not_exists DenselyOrdered
+assert_not_exists Set.Subsingleton
```

Comment:
Anything you want to add here?

Reply:
Both `Even` and `Units` are still imported by this file (via `Algebra.Group.Int`), so I don't think there is an obvious candidate. (Sadly the `Defs` files are still quite tangled...)